modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/eventpublisher/EventPublisher.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package io.ballerina.messaging.broker.core.eventpublisher;
+
+import io.ballerina.messaging.broker.core.Broker;
+import io.ballerina.messaging.broker.eventing.EventSync;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Implementation of BrokerPublisher for {@link EventSync}.
+ */
+public class EventPublisher implements EventSync {
+
+    private CorePublisher exchangePublisher = new NullCorePublisher();
```

Comment:
Ideally, we should queue the events until the broker core is initialized. We can do it in a later iteration. Let's create an issue so that we do not lose track of it.